### PR TITLE
adding label distributions to Constituent and Relation

### PR DESCRIPTION
**Pull request by mssammon**
The label distribution is presently created **on demand** (one less map entry for all cases when only a single label is specified). I.e. either you specify a map of label to score on instantiating the Constituent/Relation and it is explicit from then on, or you only specify a single label, and if someone asks for the label scores a one-entry map is created and stored and returned. 
We *could* make it explicit for single-label instantiation too. @cowchipkid what are your thoughts? 
BTW no idea why all the README changes from @Kevin5025 show up on this PR.

### CODE REVIEW

**Comment by danyaljj** [comment]
About Kevin's commits appearing here: I think you didn't checkout the latest version from `IllinoisCogComp:master`.

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/math/ArgMax.java [hunk 1]
```diff
@@ -7,6 +7,10 @@
  */
 package edu.illinois.cs.cogcomp.core.math;
 
+import com.sun.org.apache.xpath.internal.Arg;
```
Comment (bhargav) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/math/ArgMax.java]: stray import?

File: core-utilities/README.md [hunk 2]
```diff
@@ -346,51 +346,7 @@ for (int i = 0; i < ta.size(); i++) {
 }
 ```
 
-###Creating Annotators
```
Comment (danyaljj) [inline core-utilities/README.md]: @mssammon bring this back?

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/Constituent.java [hunk 3]
```diff
@@ -45,6 +47,7 @@
 
     protected final String viewName;
     protected Map<String, String> attributes;
+    protected Map<String, Double> labelsToScores;
```
Comment (bhargav) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/Constituent.java]: For Constituents and Relations, we should also update the serialization to JSON code to save the labelToScore field.

Comment (mssammon) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/Constituent.java]: good point :-)

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/Constituent.java [hunk 4]
```diff
@@ -139,6 +159,39 @@ public Constituent(String label, double score, String viewName, TextAnnotation t
                 + ", " + endCharOffset + ")";
     }
 
+    /**
+     * return map of labels to scores. If not explicitly created, generates a trivial distribution
+<<<<<<< HEAD
```
Comment (danyaljj) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/Constituent.java]: Fix these merge conflict messages

**Comment by mssammon** [comment]
@danyaljj @bhargav I think I responded to everything

**Comment by bhargav** [comment]
Couple of points
- the `hashcode()` methods need to be updated for the new field.
- the `labelToScores` can currently be modified. Should it be an immutable map; to prevent modification? This is assuming that labels (and corresponding) score are assigned when the constituent/relation is created (mostly likely in an output annotator).

**Comment by danyaljj** [comment]
Briefly skimmed through the PR. Looks good to me. 
Merge call with @bhargav

**Comment by bhargav** [comment]
Changes look good. 

@mssammon One small clarification - I see a mapDB-test.db file in the PR but it did not seem to be used anywhere.

**Pull request merged**